src/Control/Monad/Free.hs

```diff
@@ -432,3 +512,20 @@ freeDataType = mkDataType "Control.Monad.Free.FreeF" [pureConstr, freeConstr]
 {-# NOINLINE freeDataType #-}
 
 #endif
+
+-------------------------------------------------------------------------------
+-- transformers-0.4 helpers, copy from prelude-extras
+-------------------------------------------------------------------------------
+
+#ifndef LIFTED_FUNCTOR_CLASSES
+-- If Show1 and Read1 are ever derived by the same mechanism as
```

Comment:
this can be moved to separate module, to be reused in other modules. We need own `Lift1`, which uses right contexts